src/Nox.Generator/Application/Factories/EntityMapper.template.cs

```
@@ -46,29 +46,45 @@ public override void MapToEntity({{entity.Name}} entity, Entity entityDefinition
-        noxTypeValue = CreateNoxType<Nox.Types.{{attribute.Type}}>(entityDefinition,"{{attribute.Name}}",dto.{{attribute.Name}});
-        if(noxTypeValue != null)
+        noxTypeValue = CreateNoxType<Nox.Types.{{attribute.Type}}>(entityDefinition, "{{attribute.Name}}", dto.{{attribute.Name}});
+        if (noxTypeValue != null)
         {        
             entity.{{attribute.Name}} = noxTypeValue;
         }
     {{- else }}
 
         // TODO map {{attribute.Name}} {{attribute.Type}} remaining types and remove if else
-    {{- end}}        
+    {{- end }}
+    {{- end }}
+    {{ for relationship in entity.Relationships }}
+    {{- if (relationship.Relationship == "ZeroOrOne" || relationship.Relationship == "ExactlyOne") && relationship.ShouldGenerateForeignOnThisSide}}
+
+        /// <summary>
+        /// {{entity.Name}} {{relationship.Description}} {{relationship.Relationship}} {{relationship.EntityPlural}}
+        /// </summary>
+        noxTypeValue = CreateNoxType<Nox.Types.{{relationship.Related.Entity.Keys[0].Type}}>(entityDefinition, "{{relationship.Name}}", dto.{{relationship.Entity}}Id);
+        if (noxTypeValue != null)
```

In this case value of the relationship entity ID should be populated/required, so we might need to throw an Exception if the noxTypeValue is null.

---

replied above. Yes, but we don't have this for keys and attributes right now, so I think we need to have a task for implementing this, including validating that a related entity exists when specified.